app/navigators/ChargeDNavigator.scala

```diff
@@ -114,14 +98,68 @@ class ChargeDNavigator @Inject()(val dataCacheConnector: UserAnswersCacheConnect
+  }
+
+  private def routeFromIsPublicServicePensionsRemedyPage(userAnswers: UserAnswers, srn: String, startDate: LocalDate
+                                                         , accessType: AccessType, version: Int, index: Int): Call = {
+    userAnswers.get(IsPublicServicePensionsRemedyPage(ChargeTypeLifetimeAllowance, index)) match {
+      case Some(true) => controllers.mccloud.routes.IsChargeInAdditionReportedController
+        .onPageLoad(ChargeTypeLifetimeAllowance, NormalMode, srn, startDate, accessType, version, index)
+      case Some(false) => CheckYourAnswersController.onPageLoad(srn, startDate, accessType, version, index)
+      case _ => sessionExpiredPage
+    }
+  }
+
+  private def routeFromIsChargeInAdditionReportedPage(userAnswers: UserAnswers, srn: String, startDate: LocalDate
+                                                      , accessType: AccessType, version: Int, index: Int): Call = {
+    userAnswers.get(IsChargeInAdditionReportedPage(ChargeTypeLifetimeAllowance, index)) match {
+      case Some(true) => controllers.mccloud.routes.WasAnotherPensionSchemeController
+        .onPageLoad(ChargeTypeLifetimeAllowance, NormalMode, srn, startDate, accessType, version, index)
+      case Some(false) => CheckYourAnswersController.onPageLoad(srn, startDate, accessType, version, index)
+      case _ => sessionExpiredPage
+    }
+  }
+
+  private def routeFromWasAnotherPensionSchemePage(userAnswers: UserAnswers, srn: String, startDate: LocalDate
+                                                   , accessType: AccessType, version: Int, index: Int): Call = {
+    userAnswers.get(WasAnotherPensionSchemePage(ChargeTypeLifetimeAllowance, index)) match {
+      case Some(true) => controllers.mccloud.routes.EnterPstrController
+        .onPageLoad(ChargeTypeLifetimeAllowance, NormalMode, srn, startDate, accessType, version, index, countSchemeSize(userAnswers, index))
+      case Some(false) =>
+        controllers.mccloud.routes.TaxYearReportedAndPaidController
+          .onPageLoad(ChargeTypeLifetimeAllowance, NormalMode, srn, startDate, accessType, version, index, None)
+      case _ => sessionExpiredPage
+    }
+  }
+
+  private def routeFromChargeAmountReportedPage(userAnswers: UserAnswers, srn: String, startDate: LocalDate, accessType: AccessType
+                                                , version: Int, index: Int, schemeIndex: Option[Int]): Call = {
+    val schemeSize = countSchemeSize(userAnswers, index)
+    val schemeSizeLessThan5 = schemeSize > 0 && schemeSize < 5
+    (schemeIndex, schemeSizeLessThan5) match {
+      case (Some(i), true) => controllers.mccloud.routes.AddAnotherPensionSchemeController
+        .onPageLoad(ChargeTypeLifetimeAllowance, NormalMode, srn, startDate, accessType, version, index, i)
+      case (_, false) => CheckYourAnswersController.onPageLoad(srn, startDate, accessType, version, index)
```

Comment:
what about schemeSizeLessThan5 => true but schemeIndex => None.
don't you need _ case ? 
like case _ => sessionExpiredPage ?